presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/CheckUnsupportedPrestissimoTypes.java

```diff
@@ -121,71 +130,81 @@ public Void visitValues(ValuesNode node, Void context)
         public Void visitFilter(FilterNode node, Void context)
         {
             visitPlan(node, context);
-            node.getPredicate().accept(noTimeStampWithTimeZoneTypeChecker, null);
+            node.getPredicate().accept(unsupportedTypeChecker, null);
             return null;
         }
     }
 
-    private static class NoTimeStampWithTimeZoneTypeChecker
+    private static class UnsupportedTypeChecker
             extends DefaultRowExpressionTraversalVisitor<Void>
     {
         @Override
         public Void visitConstant(ConstantExpression literal, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(literal.getType()), errorMessage);
+            String errorMessage = hasType(literal.getType());
+            checkState(errorMessage == null, errorMessage);
             return null;
         }
 
         @Override
         public Void visitVariableReference(VariableReferenceExpression reference, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(reference.getType()), errorMessage);
+            String errorMessage = hasType(reference.getType());
+            checkState(errorMessage == null, errorMessage);
             return null;
         }
 
         @Override
         public Void visitInputReference(InputReferenceExpression input, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(input.getType()), errorMessage);
+            String errorMessage = hasType(input.getType());
+            checkState(errorMessage == null, errorMessage);
             return null;
         }
 
         @Override
         public Void visitCall(CallExpression call, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(call.getType()), errorMessage);
+            String errorMessage = hasType(call.getType());
+            checkState(errorMessage == null, errorMessage);
             return super.visitCall(call, context);
         }
 
         @Override
         public Void visitSpecialForm(SpecialFormExpression specialForm, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(specialForm.getType()), errorMessage);
+            String errorMessage = hasType(specialForm.getType());
+            checkState(errorMessage == null, errorMessage);
             return super.visitSpecialForm(specialForm, context);
         }
 
         @Override
         public Void visitIntermediateFormExpression(IntermediateFormExpression expression, Void context)
         {
-            checkState(!hasTimestampWithTimezoneType(expression.getType()), errorMessage);
+            String errorMessage = hasType(expression.getType());
+            checkState(errorMessage == null, errorMessage);
             return super.visitIntermediateFormExpression(expression, context);
         }
     }
 
-    private static boolean hasTimestampWithTimezoneType(Type type)
+    private static String hasType(Type type)
```

Comment:
now this function returns error message, consider rename this function

Reply:
Done